mlflow/pyfunc/__init__.py

```diff
@@ -1200,7 +1209,7 @@ def udf(
             pyfunc_backend = _get_flavor_backend(
                 local_model_path_on_executor,
                 workers=1,
-                install_mlflow=False,
+                install_mlflow=(mlflow_home is None),
```

Comment:
Previously, we would never install MLflow. Now, it seems like we'll almost always install MLflow, since users don't typically set `mlflow_home`. Is this necessary? cc @WeichenXu123

Reply:
Oh, nice catch, it's inverted. I want to install mlflow only when MLFLOW_HOME is set. MLFLOW_HOME (not very descriptive name in retrospect) is meant for development - so that you can test your local changes when running stuff in a conda environment. For example, in this case, the tests were failing because the conda environment was installing mllfow from pip and hence was expecting the old input.

Reply:
yes I think only in testing case we need set `install_mlflow` to be true